homeassistant/components/huum/climate.py

```diff
@@ -0,0 +1,131 @@
+"""Support for Huum wifi-enabled sauna."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from huum.const import SaunaStatus
+from huum.exceptions import SafetyException
+from huum.huum import Huum
+from huum.schemas import HuumStatusResponse
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import ClimateEntityFeature, HVACMode
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    PRECISION_WHOLE,
+    TEMP_CELSIUS,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Huum sauna with config flow."""
+    _huum_handler = Huum(
+        entry.data[CONF_USERNAME],
+        entry.data[CONF_PASSWORD],
+        session=async_get_clientsession(hass),
+    )
+
+    try:
+        entry_status = await _huum_handler.status_from_status_or_stop()
+    except Exception as status_error:
+        raise PlatformNotReady(status_error) from status_error
```

Comment:
Would be better to have this in `__init__.py` instead and thereby raise ConfigEntryNotReady.
Also try to use the correct exceptions instead of just broad exceptions.

Reply:
Could you expand on this a bit? This was one of the parts in the tutorial that was a bit unclear to me, the entire setup of a platform?
I mean, if I'm getting an error here, I would say that the configuration is no longer valid either and the platform can't be ready. But is that the wrong thing to do here? Or should I do this in two places?

Reply:
As it is right now it would setup the config entry but fail with the platform so you don't get any entities.
It would be better to move this to `__init__` and use ConfigEntryNotReady as then it would fail to setup the config entry and the user would be notified in a better way.
Also it will automatically retry.